lib/Parse/ParseExpr.cpp

```diff
@@ -2092,6 +2073,13 @@ ParserResult<Expr> Parser::parseExprStringLiteral() {
     Status.setIsParseError();
     return makeParserResult(Status, new (Context) ErrorExpr(Loc));
   }
+    
+  // The simple case: just a single literal segment.
+  if (Segments.size() == 1 &&
+      Segments.front().Kind == Lexer::StringSegment::Literal) {
+    return makeParserResult(
+                            createStringLiteralExprFromSegment(Context, L, Segments.front(), Loc));
+  }
```

Comment:
The block starting [here](https://github.com/apple/swift/pull/24718/files#diff-3f6a161822412eea4b0e705487dd88c1L2059) creates several AST nodes they are completely useless and discarded for non-interpolated string literal. Since majority of string literals are non-interpolated, it probably affects performance in both CPU and memory. Could you move this block to before that? I'm OK to duplicate `SyntaxContext` related work for that.

Reply:
Addressed your feedback. Hope it has less AST nodes now